### PR TITLE
fixes issue #42

### PR DESCRIPTION
**Pull request by aroidzap**

### CODE REVIEW

File: grapher/Models/Calculations/AccelCalculator.cs [hunk 1]
```diff
@@ -99,6 +99,7 @@ public void Calculate(AccelChartData data, ManagedAccel accel, double starter, I
                 }
 
                 var ratio = outMagnitude / simulatedInputDatum.velocity;
+                ratio = Math.Min(int.MaxValue, Math.Max(int.MinValue, ratio));
```
Comment (JacobPalecki) [inline grapher/Models/Calculations/AccelCalculator.cs:102]: Can you please share with me a settings.json where the later conversion of this ratio to lower precision causes overflow?

Comment (JacobPalecki) [reply grapher/Models/Calculations/AccelCalculator.cs:102]: Wondering as well why you are only guarding ratio here and not slope.

File: grapher/Models/Calculations/AccelCalculator.cs [hunk 2]
```diff
@@ -203,6 +204,8 @@ public void CalculateCombinedDiffSens(AccelData data, ManagedAccel accel, Driver
 
                 var xRatio = settings.sensitivity.x * ratio;
                 var yRatio = settings.sensitivity.y * ratio;
+                xRatio = Math.Min(int.MaxValue, Math.Max(int.MinValue, xRatio));
+                yRatio = Math.Min(int.MaxValue, Math.Max(int.MinValue, yRatio));
```
Comment (JacobPalecki) [inline grapher/Models/Calculations/AccelCalculator.cs:208]: Can we make this a helper named something like "ClampToInt"?

File: grapher/Models/Calculations/AccelCalculator.cs (hunk 1 above)
Comment (aroidzap) [reply grapher/Models/Calculations/AccelCalculator.cs:102]: it crashed when use default configuration and: 
x=0, y>0 or x>0, y=0

Comment (aroidzap) [reply grapher/Models/Calculations/AccelCalculator.cs:102]: Actually this hotfix isn't the best, but it solves theese two extreme cases. I'm not even sure, that clamping to int isn't too much. The problem is actually in drawing data. Clamped should be everything, what goes into DataBind, DataBindX, DataBindXY, SetMinMaxX, etc... (something what could be converted to Decimal, without any exception). Interesting thing also is, that you could pass very low value, but after you pass 0, it breaks - or shows very big range on axes with this fix. Maybe there is some other part of code, which sets something to NaN, Infinity or very big or small value.

File: grapher/Models/Calculations/AccelCalculator.cs (hunk 2 above)
Comment (aroidzap) [reply grapher/Models/Calculations/AccelCalculator.cs:208]: Maybe, but unless we even know, clamping to int really makes sense (not that it just works for these two extreme cases), it's not worth polishing.

**Review by a1xd — APPROVED** [review]

File: grapher/Models/Calculations/AccelCalculator.cs (hunk 1 above)
Comment (JacobPalecki) [reply grapher/Models/Calculations/AccelCalculator.cs:102]: Closing this as the issue is no longer seen.

**Pull request closed without merging**